testnet/stacks-node/src/burnchains/commitment.rs

```diff
@@ -98,6 +104,98 @@ fn l1_get_nonce(l1_rpc_interface: &str, address: &StacksAddress) -> Result<u64,
     Ok(response_json.nonce)
 }
 
+/// Compute an effective fee to use, based on a transaction, and response scalars. Use the equation:
+///     `base_fee` + `fee_rate` x `cost_scalar_change_by_byte` x (`final_size` - `estimated_size`)
+pub fn calculate_fee_rate_adjustment(
+    transaction: &StacksTransaction,
+    base_fee: u64,
+    fee_rate: f64,
+    cost_scalar_change_by_byte: f64,
+) -> Result<u64, FeeCalculationError> {
+    let mut transaction_bytes = vec![];
+    transaction
+        .consensus_serialize(&mut transaction_bytes)
+        .map_err(|e| {
+            warn!("Error serializing transaction: {:?}", transaction);
+            FeeCalculationError::ErrorSerializingTransaction
+        })?;
```

Comment:
`StacksMessageCodec` defines `serialize_to_vec` that you should use instead for this:

```suggestion
    let transaction_bytes = transaction.serialize_to_vec();
```